src/dbnode/encoding/types.go

```diff
@@ -265,7 +274,7 @@ type NewDecoderFn func() Decoder
 type EncoderAllocate func() Encoder
 
 // ReaderIteratorAllocate allocates a ReaderIterator for a pool.
-type ReaderIteratorAllocate func(reader io.Reader) ReaderIterator
+type ReaderIteratorAllocate func(reader io.Reader, descr namespace.SchemaDescr) ReaderIterator
```

Comment:
@prateek @richardartoul this is another breaking change that bit us in statsdex. Would it make sense to take a look at these types and move those that are meant to be exported to the `x` package and those that aren't to an `internal` package to ensure we can't import them?

Reply:
What was statsdex using it for?

Reply:
@arnikola added support to query for encoding cross zone responses in the query service using m3ts... Though that kinda misses the more general point of should this be exposed at all :)

Reply:
@jeromefroe I don't think moving this would help. If you're using the encoding package to encode data, then the code is going to break when we make breaking changes to the encoders.

I can't make the constructor function for the encoding package private